pkg/enmasse/service.go

```diff
@@ -29,37 +28,78 @@ const (
 
 type Service struct {
 	k8sclient   kubernetes.Interface
-	routeClient routev1.RouteInterface
-	ns          string
+	routeClient routev1.RoutesGetter
+	enmasseNS   string
+	httpClient  *http.Client
 }
 
-func NewService(k8sclient kubernetes.Interface, routeClient routev1.RouteInterface, targetNS string) *Service {
-	return &Service{k8sclient: k8sclient, routeClient: routeClient, ns: targetNS}
+func NewService(k8sclient kubernetes.Interface, routeClient routev1.RoutesGetter, httpClient *http.Client, targetNS string) *Service {
+	return &Service{k8sclient: k8sclient, routeClient: routeClient, enmasseNS: targetNS, httpClient: httpClient}
+}
+
+func (s *Service) DeleteUser(userName, realm string) error {
+	route, err := s.routeClient.Routes(s.enmasseNS).Get("keycloak", metav1.GetOptions{})
```

Comment:
is adding / deleting users something more services might need? Should we make the interaction with Keycloak an external package that could be used by multiple Integrations?